packages/client/src/features/change-bounds/resize-tool.ts

```diff
@@ -0,0 +1,157 @@
+    static ID = 'glsp.resize-keyboard';
+
+    isEditTool = true;
+
+    @inject(KeyTool) protected readonly keytool: KeyTool;
+    @inject(TYPES.IMovementRestrictor) @optional() readonly movementRestrictor?: IMovementRestrictor;
+    @inject(TYPES.ISnapper) @optional() readonly snapper?: ISnapper;
+
+    protected resizeKeyListener: ResizeKeyListener;
+
+    get id(): string {
+        return ResizeTool.ID;
+    }
+
+    enable(): void {
+        this.resizeKeyListener = new ResizeKeyListener(this);
+        this.keytool.register(this.resizeKeyListener);
+    }
+
+    disable(): void {
+        this.keytool.deregister(this.resizeKeyListener);
+    }
+}
+@injectable()
+export class ResizeKeyListener extends KeyListener {
+    protected grid = { x: 20, y: 20 };
+    protected isEditMode = false;
+
+    constructor(protected readonly tool: ResizeTool) {
+        super();
+
+        if (this.tool.snapper instanceof GridSnapper) {
+            this.grid = this.tool.snapper?.grid;
+        }
+    }
+
+    override keyDown(element: SModelElement, event: KeyboardEvent): Action[] {
+        const actions: Action[] = [];
+
+        if (this.isEditMode && matchesKeystroke(event, 'Escape')) {
+            this.isEditMode = false;
+        }
+
+        if (this.getSelectedElements(element.root).length > 0) {
+            if (!this.isEditMode && matchesKeystroke(event, 'KeyR', 'alt')) {
+                this.isEditMode = true;
+            }
+
+            if (this.isEditMode) {
+                const selectedElements = this.getSelectedElements(element.root);
+                if (event.key === '+') {
+                    for (const elem of selectedElements) {
+                        const action = this.handleResizeElement(elem, this.grid.x, this.grid.y);
+                        if (action) {
+                            actions.push(action);
+                        }
+                    }
+                } else if (matchesKeystroke(event, 'Minus')) {
+                    for (const elem of selectedElements) {
+                        const action = this.handleResizeElement(elem, -this.grid.x, -this.grid.y);
+                        if (action) {
+                            actions.push(action);
+                        }
+                    }
+                } else if (matchesKeystroke(event, 'Digit0', 'ctrl')) {
+                    for (const elem of selectedElements) {
+                        elem.bounds = { x: elem.bounds.x, y: elem.bounds.y, width: minWidth(elem), height: minHeight(elem) };
+                        const action = this.handleResizeElement(elem, 0, 0);
+                        if (action) {
+                            actions.push(action);
+                        }
+                    }
+                }
+            }
+        }
+
+        return actions;
+    }
+
+    protected handleResizeElement(element: SModelElement & BoundsAware, deltaWidth: number, deltaHeight: number): Action | undefined {
```

Comment:
```suggestion
    protected handleResizeElement(element: SelectableBoundsAware, deltaWidth: number, deltaHeight: number): Action | undefined {
```

I think it would be nice to use the `SelectableBoundsAware` everywhere instead of the combined type.